### ISSUE TITLE
Add a ladder material?

### ISSUE THREAD

**Issue by IAmNotHanni** [inexor-game/code#135]
At the moment we can only make ladders by stacking shaped cubes on each other.
Why not add a ladder material and a third person view when climbing a ladder?

**Comment by Croydon** [comment]
Sounds great, but as multiple times noted, the way how materials are implemented needs refactoring. So we should perhaps wait for this. The biggest point is to eleminating the need for such workarounds as water, water2, water3... instead make it for map builders possible to define own attributes for a basic materials. Should be unlimited, so somebody can have as many different "waters" (or lavas or whatever) as one whishs. Perhaps make it even possible to define completely own materials, however I suppose that's kind of complicated.

**Comment by koraa** [comment]
Agreed. I'd like to see even more complex materials like fogs or something, that can be mutated as a cellular automate+muparser or something.

Though this case is a bit different, since it is a material for textures. This might be implemented using the 3D material system, but we should definitely provide a specific interface for that:
* Using texture painting to define the values of texture materials
* Using actual textures, possibly loaded as part of the material to set those. This would be even more useful with an ingame texture editor.

And, just in order to have them written down somewhere, here are my suggestions for texture materials:
* *friction* – Factor.  x<0 = Increasing acceleration; 0 = No friction; 1 = Normal, x>1 = Very high friction
* *slipperiness* – 0 = Normal control over movement; x>0 = Increasingly less control over movement
* *bouncyness* – Involuntary jump on collision expressed as a factor of falling height; possibly with an absolute minimum/maximum value
* *trampoline* – Same as bounciness, but only applies to voluntary jumps
* *healt* – Factor with minimum/maximum added/reduced health per second

Some attributes (like health) could even be expressed by using a generic system with muparser equations, executed on collision and contact.

**Comment by aschaeffer** [comment]
Further "material" issues:

#65 
#112
